### PR TITLE
Implement dual attack and dual weapons + attack all fix

### PR DESCRIPTION
**Pull request by fmatthew5876**

### CODE REVIEW

**Comment by fmatthew5876** [comment]
This one is ready now too.

**Comment by fmatthew5876** [comment]
2 things I need to verify:

1. What happens if attack damage heals a state like confuse but then the weapon inflicts it at the same time?
2. What happens if weapon1 inflicts a state and weapon2 heals that very same state?

**Comment by fmatthew5876** [comment]
I've tested all the state behavior and it matches `RPG_RT`. Now this really is ready.

**Comment by carstene1ns** [comment]
jenkins: test this please (newlib check).

**Comment by carstene1ns** [comment]
test this please

**Comment by fmatthew5876** [comment]
rebased to master

File: src/scene_battle_rpg2k.cpp [hunk 1]
```diff
@@ -862,19 +862,29 @@ void Scene_Battle_Rpg2k::SelectNextActor() {
 			break;
 	}
 
-	if (random_target || auto_battle || active_actor->GetAutoBattle()) {
-		if (!random_target) {
-			random_target = Main_Data::game_enemyparty->GetRandomActiveBattler();
-		}
-
-		// ToDo: Auto battle logic is dumb
```
Comment (Ghabry) [inline src/scene_battle_rpg2k.cpp]: What I meant with "logic is dumb" is that the RPG_RT auto battle can also invoke skills.
Though I'm not a fan of auto battle wasting magic.

Comment (fmatthew5876) [reply src/scene_battle_rpg2k.cpp]: For the auto battle command it's a matter of debate. For actors with auto battle set, they should use skills.

File: src/game_battlealgorithm.h [hunk 2]
```diff
@@ -187,6 +186,15 @@ class AlgorithmBase {
 	 * @return Battle Animation or NULL if no animation is assigned
 	 */
 	const RPG::Animation* GetAnimation() const;
+	const RPG::Animation* GetSecondAnimation() const;
+
+	/**
+	* Checks if the animation has already played once
```
Comment (Ghabry) [inline src/game_battlealgorithm.h]: That's a typical Albeleon lack-of-space doc-comment

File: src/game_battlealgorithm.h [hunk 3]
```diff
@@ -367,6 +376,11 @@ class AlgorithmBase {
 	 */
 	RPG::State::Restriction GetSourceRestrictionWhenStarted() const;
 
+	/*
```
Comment (Ghabry) [inline src/game_battlealgorithm.h]: And this lacks one *

**Comment by fmatthew5876** [comment]
comments fixed

File: src/game_battlealgorithm.h (hunk 2 above)
Comment (Ghabry) [reply src/game_battlealgorithm.h]: With lack-of-space I meant that there is a single space missing before the *, Albeleon did this the whole time.

**Comment by Ghabry** [comment]
Well, the animations are not playing at the same time as in RPG_RT but after each other. Is that okay for now?

**Comment by Ghabry** [comment]
Okay this is a bit obscure, in Unterwegs in Düsterburg I get two animations in RPG_RT but in other games I get both animations interleaved. Can be investigated later.

**Pull request merged**